### PR TITLE
Implemented new particle diagnostics in picmi

### PR DESCRIPTION
**Pull request by dpgrote**
This PR updates the picmi interface to use the new format for diagnostics for species.

### CODE REVIEW

**Comment by MaxThevenet** [comment]
Thanks for this PR! Could you have a look at the LGTM alert?

**Comment by dpgrote** [comment]
The LGTM alert should now be fixed.

**Comment by MaxThevenet** [comment]
Oh, it seems that the TravisCI tests are failing on this branch

**Comment by MaxThevenet** [comment]
Could you merge master in this branch. I still don't understand why tests fail, and I cannot push on your fork.

File: Python/pywarpx/Diagnostics.py [hunk 1]
```diff
@@ -6,6 +6,20 @@
...
+class Diagnostic(Bucket):
+    """
+    This is the same as a Bucket, but checks that any attributes are always given the same value.
+    """
+    def add_new_attr_with_check(self, name, value):
+        if name.startswith('_'):
+            self._localsetattr(name, value)
+        else:
+            if name in self.argvattrs:
+                assert value == self.argvattrs[name], Exception(f'Diagnostic attributes not consistent for {self.instancename}')
```
Comment (LDAmorim) [inline Python/pywarpx/Diagnostics.py:20]: I don't know if `f` is a variable, but I thought the `assert` in picmi using `Exception` doesn't need the `f`.
Only the string.
```suggestion
                assert value == self.argvattrs[name], Exception('Diagnostic attributes not consistent for {self.instancename}')
```

Comment (dpgrote) [reply Python/pywarpx/Diagnostics.py:20]: Hi @LDAmorim The f-string is a new, very nice, way in Python of formatting strings. Here is the documentation https://docs.python.org/3/reference/lexical_analysis.html#index-24

Comment (LDAmorim) [reply Python/pywarpx/Diagnostics.py:20]: Thanks @dpgrote ! This is really nice indeed 👍

File: .travis.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@
 dist: xenial
 language: c++
 sudo: true
-cache: pip
+#cache: pip
```
Comment (LDAmorim) [inline .travis.yml:11]: Thanks for fixing this!

**Pull request merged**